src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+
+import { getUser } from '../../utils';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  addTodo: (todo: Todo) => void;
+  users: User[];
+}
+
+export const TodoForm: React.FC<Props> = ({ addTodo, users }) => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const restart = () => {
+    setTitle('');
+    setUserId(0);
+    setHasUserError(false);
+    setHasTitleError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (userId === 0) {
+      setHasUserError(true);
+    }
+
+    if (!title || !title.trim()) {
+      setHasTitleError(true);
+    }
+
+    if (title.trim() && userId) {
+      addTodo({
+        id: 0,
+        title: title.trim(),
+        completed: false,
+        userId,
+        user: getUser(userId),
+      });
+
+      restart();
+    }
```

Comment:
```suggestion
      addTodo({
        title: title.trim(),
        completed: false,
        userId,
        user: getUser(userId),
      });

      restart();
```

Reply:
now we can avoid additional if statement since we have already checked title and user

Reply:
we don't need to add some id here since we'll do it in app component